### PR TITLE
files for new funtion oi_bicycle_parking, for #107

### PR DESCRIPTION
**Pull request by hulsiejames**
New function to obtain bycycle parking infrastructure from OSM data. 

This does require networks to contain Node features rather than just lines, as the majority of these are tagged amenity="bicycle_parking". This means that creating the transport infrastrcutre data packs will take longer as we also need to extract "points" info from the .osm.pbf file. 

However, now that we have data on node information we can now assess this (thinking crossings with traffic lights, and other pois useful for active travel)

### CODE REVIEW

File: R/oi_bicycle_parking.R [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#' Identify for bicycle parking.
```
Comment (Robinlovelace) [inline R/oi_bicycle_parking.R:1]: I suggest:

```
#' Identify bicycle parking places from OSM
```

There was a slight typo in the original and no need for a full stop.

File: R/oi_bicycle_parking.R [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#' @return - The provided `osm_sf` is returned with an additional column, 
+#' `oi_cycle_parking` containing the value `"yes"` if this feature allows 
+#' bicycle parking.
+#' @details - Note: the input `osm_sf` must containt the following tags:
+#' `c("amenity")`. Note that as most (if not all) amenity tags for bicycle 
+#' parking are assigned to nodes, the `osm_sf` must also contain nodes. (the
+#' layer `"points"` must be requested when using `osmextract::oe_get()`)
+#' @export oi_bicycle_parking 
+#' @examples
+#' internal_data = example_data
```
Comment (Robinlovelace) [inline R/oi_bicycle_parking.R:21]: Are these OSM ways?

Most parking spaces are points I think. Would like to see some stats on bicycle parking in different layers.

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Small change requested and a question.

File: R/oi_bicycle_parking.R (hunk 2 above)
Comment (hulsiejames) [reply R/oi_bicycle_parking.R:21]: For now, the eaxmple data is ways only. I added the amentity column with mutate so the checks (for now) would pass - I am going to update the exmaple data to contain nodes too. 

>Most parking spaces are points I think. Would like to see some stats on bicycle parking in different layers.

Indeed this is the case -  for a 2.5km circular buffer around leeds city centre there are 291 amenity=="bicycle_parking" tags, all belonging to Nodes.

I think worldwide - there are ~ 950 ways that contain amenity="bicycle_parking" (add taginfo link here). So, my current plan is to combine the "lines" & "points" layers into a single combined network for each LAD, and then create the data packs from this, so we can obtain bicycle parking features (and any other useful nodes)

**Review by hulsiejames — COMMENTED** [review]
looks good

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
> My current plan is to combine the "lines" & "points" layers into a single combined network for each LAD, and then create the data packs from this, so we can obtain bicycle parking features (and any other useful nodes)

I advise against this. The data packs will be more useful if they contain distinct layers. I think we need a clearly defined points layer example. Which subset of all possible points are useful for transport planning? Those are the ones we use.

I don't think combining points and lines into a single layer is a good plan, they are separate on OSM and should be separate in the data packs. The cycle parking points could be a completely separate layer, e.g. called `cycle_parking_points.gpkg` for each region.

**Comment by hulsiejames** [comment]
>I don't think combining points and lines into a single layer is a good plan, they are separate on OSM and should be separate in 
>the data packs. The cycle parking points could be a completely separate layer, e.g. called cycle_parking_points.gpkg for each >region.

I agree with this. 

New idea: 
 - Create two package example data sets, both a small raidus around leeds city centre. One a "lines" layer of ways, the other a "points" layer of nodes.
 - For each LAD create two data packs - one for each unique layer being analysed (points and lines for now) and to specify the layer in datapack file names (i.e. `26_08_2022_datapack_lines_Leeds.geojson`)

**Review by hulsiejames — COMMENTED** [review]
all edits done

File: R/oi_bicycle_parking.R [hunk 3]
```diff
@@ -18,9 +18,8 @@
 #' layer `"points"` must be requested when using `osmextract::oe_get()`)
 #' @export oi_bicycle_parking 
 #' @examples
-#' internal_data = example_data
-#' #example_data = example_data %>% dplyr::mutate(amenity = NA)
-#' example_output = oi_bicycle_parking(example_data)
+#' data = example_data_pois
```
Comment (Robinlovelace) [inline R/oi_bicycle_parking.R:21]: Nice, yes that solves it as far as the docs go I think.

**Review by Robinlovelace — APPROVED** [review]
Looking good. May be worth a conversation about how to present the outputs: should the output be just bike parking points rather than a new column?

In any case the function itself looks good and happy for it to be merged.

**Comment by hulsiejames** [comment]
> Looking good. May be worth a conversation about how to present the outputs: should the output be just bike parking points rather than a new column?
> 
> In any case the function itself looks good and happy for it to be merged.

I agree a meeting conversation would be good - I can arrange one for when you are back from holiday.


(My thoughs to discuss later)
I like the idea of a single bicycle_parking file/layer - but if these files are to be read into another editor by LADs (QGIS I imagine?) can they support reading multiple files of points (i.e. a bicycle_parking file, a zebra_corssings file, etc.), or would it be better to have all point info in a file and this can be filtered within the editor? 

I'll merge this now and close the branch.

**Comment by Robinlovelace** [comment]
> I like the idea of a single bicycle_parking file/layer - but if these files are to be read into another editor by LADs (QGIS I imagine?) can they support reading multiple files of points (i.e. a bicycle_parking file, a zebra_corssings file, etc.), or would it be better to have all point info in a file and this can be filtered within the editor?

Fewer files is good so I'm erring towards including multiple types of point in a points layer.

**Pull request merged**